packages/app-store/src/appStore.ts

```diff
@@ -60,6 +60,32 @@ export type CustomSetState = (
 
 export type SetterCreator<K extends keyof AppState> = (setState: CustomSetState) => AppState[K]
 
+/**
+ * Cleans up title field in save list items by extracting the filename without extension
+ * when the title is actually a full path
+ */
+const cleanupSaveListTitles = (saveList: SaveItem[]): SaveItem[] => {
+  let cleanedList = saveList.map((item) => {
+    // Check if the title looks like a path
+    if (item.title && (item.title.includes('/') || item.title.includes('\\'))) {
+      // Extract filename without extension from the path
+      const filename = item.title.split(/[\/\\]/).pop() || item.title
+      // Remove extension if present
+      const titleWithoutExt = filename.replace(/\.[^/.]+$/, '')
+      return { ...item, title: titleWithoutExt }
+    }
+    return item
+  })
+  // remove paths that do not exist anymore
+  // this doesn't actually work yet, because I couldn't figure out how to get fs on the global object before the store is created
+  const fs = window['__HEDRON'].fs
```

Comment:
I tried getting it to add fs to the __HEDRON object in the desktop package, but I think this was always running first? Regardless, even trying to add it after the window was created, the object never displayed fs on it, despite using the same code as in globalVars (blend existing object, don't overwrite)

Reply:
This will never work as fs is not available to the Electron renderer. I think we need to build a little middle layer to allow users to save files.